partcad/src/partcad/converters/stl_converter.py

```
@@ -0,0 +1,100 @@
+        - ascii: Boolean flag to determine if STL should be in ASCII format.
+        """
+        logging.info(f"Attempting to write STL file: {self.output_file}")
+
+        # Check if the shape is valid
+        analyzer = BRepCheck_Analyzer(shape)
+        if not analyzer.IsValid():
+            raise RuntimeError(f"The shape is not valid and cannot be exported to STL: {shape}")
+
+        # Check if the shape is solid
+        if not is_solid(shape):
+            raise RuntimeError("The shape is not a solid and cannot be exported to STL.")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)